### PR TITLE
#341 Rename main branch

### LINKED ISSUE #341: Rename develop branch to main
Projects nowadays use `main` as default branch. To avoid confusion we should rename the `develop` branch to `main` and delete `master`.

- [ ] Update documentation
- [ ] Add changelog entry
- [ ] Update branch protection rules in GitHub
- [ ] Rename `develop` branch to `main`
- [ ] Rename default branch in sonarcloud
- [ ] Delete `master`

### PR DESCRIPTION
**Pull request by kaklakariada**
Closes #341
This also updates the instructions for self-tracing.

### CODE REVIEW

File: doc/CHANGELOG.md [hunk 1]
```diff
@@ -25,17 +40,6 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
-  - Added support for tracing TypeScript files (`.ts`)
-
-### Continuous Integration Build
-
-- Added builds for macOS and Windows [#340](https://github.com/itsallcode/openfasttrace/issues/340)
-
-### Bugfixes
-
-- Fixed running unit tests under macOS [#308](https://github.com/itsallcode/openfasttrace/issues/308) / [PR #339](https://github.com/itsallcode/openfasttrace/pull/339)
-  - Also fixed Jacoco code coverage and integration tests
```
Comment (kaklakariada) [inline doc/CHANGELOG.md:38]: Changes where assigned to the wrong version

File: .vscode/extensions.json [hunk 2]
```diff
@@ -0,0 +1,5 @@
+{
+	"recommendations": [
+		"vscjava.vscode-java-pack"
+	],
```
Comment (kaklakariada) [inline .vscode/extensions.json:4]: This will recommend installing the correct plugin when opening OFT with VSCode

File: doc/developer_guide.md [hunk 3]
```diff
@@ -43,7 +43,6 @@ The list below shows all build time dependencies in alphabetical order. Note tha
 | [JUnit](https://junit.org/junit5)                                            | Unit testing framework                                 | Eclipse Public License 1.0    |
 | [Mockito](http://site.mockito.org/)                                          | Mocking framework                                      | MIT License                   |
 | [JUnit5 System Extensions](https://github.com/itsallcode/junit5-system-extensions) | JUnit extension for testing `System.x` calls    | Eclipse Public License 2.0     |
-| [Pitest](http://pitest.org/)                                                 | Mutation testing                                       | Apache License 2.0            |
```
Comment (kaklakariada) [inline doc/developer_guide.md:46]: Was never used

File: doc/developer_guide.md [hunk 4]
```diff
@@ -61,7 +60,8 @@ If you want to build OFT:
 
 * `git clone https://github.com/itsallcode/openfasttrace.git`
 * Run `mvn test` to run unit tests.
-* Run `mvn exec:java@trace` to run requirements tracing.
+* Run `mvn verify` to run integration tests.
+* Run `./oft-self-trace.sh` to run requirements tracing.
```
Comment (kaklakariada) [inline doc/developer_guide.md:64]: `exec:java@trace` did not work any more -> removed plugin

**Review by redcatbear — APPROVED** [review]

**Pull request merged**